typescript/src/codeActions/custom/renameToTypeParameter.ts

```diff
@@ -0,0 +1,36 @@
+import { CodeAction } from '../getCodeActions'
+import { getChangesTracker } from '../../utils'
+
+export default {
+    id: 'renameToTypeParameter',
+    name: 'Rename to Type Parameter',
+    kind: 'refactor.rewrite.renameToTypeParameter',
+    tryToApply(sourceFile, position, range, node, formatOptions, languageService) {
+        if (!node || !position) return
+        if (!ts.isIdentifier(node) || !ts.isParameter(node.parent) || !ts.isFunctionLike(node.parent.parent)) return
+
+        const functionParameters = node.parent.parent.parameters
+        const parameterIndex = functionParameters.indexOf(node.parent)
+
+        const functionDecl = node.parent.parent.parent
+        const typeChecker = languageService.getProgram()!.getTypeChecker()
+        const typeAtLocation = typeChecker.getTypeAtLocation(functionDecl)
+
+        const typeDecl = typeAtLocation.getSymbol()?.declarations?.[0]
+        if (!typeDecl || !ts.isFunctionLike(typeDecl)) return
+        const paramName = node.getText()
+        const typeParamName = typeDecl.parameters[parameterIndex]!.name.getText()
+        if (paramName === typeParamName) return
+        if (!formatOptions) return true
+        const changesTracker = getChangesTracker({})
+        changesTracker.replaceNodeWithText(sourceFile, node, typeParamName)
```

Comment:
Did you check is it going to preserve comments? eg `const a = (/*@type{...}*/test) => {}`

Reply:
Nevermind, this would lead to broken code anyway. We need to use safe rename functionality instead (but you need to call it second on applying edits of course): https://github.com/zardoy/typescript-vscode-plugins/blob/240f2e1cc89f871d2442d5b28270b38dcd51ab2f/typescript/src/decorateEditsForFileRename.ts#L17